src/Orleans/GrainDirectory/IGrainDirectory.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Orleans.Runtime;
+
+namespace Orleans.GrainDirectory
+{
+
+    /// <summary>
+    /// Recursive distributed operations on grain directories.
+    /// Each operation may forward the request to a remote owner, increasing the hopcount.
+    /// </summary>
+    interface IGrainDirectory
```

Comment:
This interface is used both in an RPC-able and a non-RPC-able context.
It is extended by IRemoteGrainDirectory, which is ISystemTarget, and by ILocalGrainDirectory, which is not.